File: frontend/src/lib/api/ai.js
import { del, get, post, put } from './core.js';

export const ai = {
  status: () => get('/ai/status'),
  planMyDay: (connectionId) =>
    get(`/ai/plan-my-day${connectionId ? `?connection_id=${connectionId}` : ''}`),
  planMyDayPreview: () => get('/ai/plan-my-day?preview=true'),
  catchMeUp: (itemId) => post(`/ai/items/${itemId}/catch-me-up`),
  findSimilar: (itemId) => post(`/ai/items/${itemId}/find-similar`),
  decompose: (itemId) => post(`/ai/items/${itemId}/decompose`),
  generateReleaseNotes: (milestoneId, connectionId) =>
    post(
      `/ai/milestones/${milestoneId}/generate-release-notes${connectionId ? `?connection_id=${connectionId}` : ''}`
    ),
  summarizeTestPlanDescription: (testSetId, connectionId) =>
    post(
      `/ai/test-sets/${testSetId}/summarize-description${connectionId ? `?connection_id=${connectionId}` : ''}`
    ),
  analyzeDependencies: (iterationId, body = {}, connectionId) =>
    post(
      `/ai/iterations/${iterationId}/analyze-dependencies${connectionId ? `?connection_id=${connectionId}` : ''}`,
      body
    ),
  acceptDependencies: (iterationId, suggestions) =>
    post(`/ai/iterations/${iterationId}/accept-dependencies`, { suggestions }),
  chat: (message, connectionId, history, context) =>
    post('/ai/chat', {
      message,
      ...(connectionId ? { connection_id: connectionId } : {}),
      ...(history?.length ? { history } : {}),
      ...(context && Object.keys(context).length ? { context } : {}),
    }),
  dailyBriefing: () => get('/ai/daily-briefing'),
};

export const aiFeatures = {
  getConfig: () => get('/admin/ai-features'),
  updateConfig: (data) => put('/admin/ai-features', data),
};

export const llmConnections = {
  getAll: () => get('/admin/llm-connections'),
  get: (id) => get(`/admin/llm-connections/${id}`),
  create: (data) => post('/admin/llm-connections', data),
  update: (id, data) => put(`/admin/llm-connections/${id}`, data),
  delete: (id) => del(`/admin/llm-connections/${id}`),
  test: (id) => post(`/admin/llm-connections/${id}/test`),
};

export const llmProviders = {
  getProviders: () => get('/llm/providers'),
  getEnabled: () => get('/llm/connections'),
};

export const actionCapabilities = {
  getAll: () => get('/admin/action-capabilities'),
  get: (id) => get(`/admin/action-capabilities/${id}`),
  create: (data) => post('/admin/action-capabilities', data),
  update: (id, data) => put(`/admin/action-capabilities/${id}`, data),
  delete: (id) => del(`/admin/action-capabilities/${id}`),
  // Workspace-scoped picker list — returns enabled capabilities the workspace
  // can reference (applies-to-all OR explicitly scoped). Optional type filter.
  getForWorkspace: (workspaceId, type) => {
    const qs = type ? `?type=${encodeURIComponent(type)}` : '';
    return get(`/workspaces/${workspaceId}/action-capabilities${qs}`);
  },
};

// actionCredentials: workspace-aware credential store referenced by HTTP
// capabilities. The plaintext secret travels only on create/rotate; every
// response is the sanitized DTO (has_secret + prefix; no ciphertext).
export const actionCredentials = {
  // Global (workspace_id IS NULL) credentials — system-admin only.
  getAllGlobal: () => get('/admin/action-credentials'),
  createGlobal: (data) => post('/admin/action-credentials', data),
  updateGlobal: (id, data) => put(`/admin/action-credentials/${id}`, data),
  rotateGlobal: (id, secret) => post(`/admin/action-credentials/${id}/rotate`, { secret }),
  deleteGlobal: (id) => del(`/admin/action-credentials/${id}`),

  // Workspace-scoped credentials — gated by action.credential.manage. The
  // list endpoint also returns globals (workspace_id null) so a single call
  // populates the credential picker.
  getForWorkspace: (workspaceId) => get(`/workspaces/${workspaceId}/action-credentials`),
  createForWorkspace: (workspaceId, data) =>
    post(`/workspaces/${workspaceId}/action-credentials`, data),
  updateForWorkspace: (workspaceId, id, data) =>
    put(`/workspaces/${workspaceId}/action-credentials/${id}`, data),
  rotateForWorkspace: (workspaceId, id, secret) =>
    post(`/workspaces/${workspaceId}/action-credentials/${id}/rotate`, { secret }),
  deleteForWorkspace: (workspaceId, id) =>
    del(`/workspaces/${workspaceId}/action-credentials/${id}`),
};

File: internal/routes/ai.go
package routes

import "net/http"

// RegisterAIRoutes registers AI-powered feature routes.
func RegisterAIRoutes(deps *Deps) {
	api := deps.API
	auth := deps.AuthMiddleware.RequireAuth
	admin := deps.PermissionMiddleware.RequireSystemAdmin()

	// AI feature endpoints (user) - rate limited to protect expensive LLM calls
	api.HandleH("GET /ai/status", auth(http.HandlerFunc(deps.AI.AI.Status)))
	api.HandleH("POST /ai/chat", auth(deps.AIRateLimiter.Limit(http.HandlerFunc(deps.AI.AI.Chat))))
	api.HandleH("GET /ai/daily-briefing", auth(http.HandlerFunc(deps.AI.AI.GetDailyBriefing)))
	api.HandleH("GET /ai/plan-my-day", auth(deps.AIRateLimiter.Limit(http.HandlerFunc(deps.AI.AI.PlanMyDay))))
	api.HandleH("POST /ai/items/{id}/catch-me-up", auth(deps.AIRateLimiter.Limit(http.HandlerFunc(deps.AI.AI.CatchMeUp))))
	api.HandleH("POST /ai/items/{id}/find-similar", auth(deps.AIRateLimiter.Limit(http.HandlerFunc(deps.AI.AI.FindSimilarItems))))
	api.HandleH("POST /ai/items/{id}/decompose", auth(deps.AIRateLimiter.Limit(http.HandlerFunc(deps.AI.AI.DecomposeItem))))
	api.HandleH("POST /ai/milestones/{id}/generate-release-notes", auth(deps.AIRateLimiter.Limit(http.HandlerFunc(deps.AI.AI.GenerateReleaseNotes))))
	api.HandleH("POST /ai/iterations/{id}/analyze-dependencies", auth(deps.AIRateLimiter.Limit(http.HandlerFunc(deps.AI.AI.AnalyzeDependencies))))
	api.HandleH("POST /ai/iterations/{id}/accept-dependencies", auth(http.HandlerFunc(deps.AI.AI.AcceptDependencies)))
	api.HandleH("POST /ai/test-sets/{id}/summarize-description", auth(deps.AIRateLimiter.Limit(http.HandlerFunc(deps.AI.AI.SummarizeTestPlanDescription))))

	// LLM provider info (user)
	api.HandleH("GET /llm/providers", auth(http.HandlerFunc(deps.AI.LLMConnection.GetProviders)))
	api.HandleH("GET /llm/connections", auth(http.HandlerFunc(deps.AI.LLMConnection.GetEnabledConnections)))

	// LLM connection management (admin)
	api.HandleH("GET /admin/llm-connections", admin(http.HandlerFunc(deps.AI.LLMConnection.ListConnections)))
	api.HandleH("POST /admin/llm-connections", admin(http.HandlerFunc(deps.AI.LLMConnection.CreateConnection)))
	api.HandleH("GET /admin/llm-connections/{id}", admin(http.HandlerFunc(deps.AI.LLMConnection.GetConnection)))
	api.HandleH("PUT /admin/llm-connections/{id}", admin(http.HandlerFunc(deps.AI.LLMConnection.UpdateConnection)))
	api.HandleH("DELETE /admin/llm-connections/{id}", admin(http.HandlerFunc(deps.AI.LLMConnection.DeleteConnection)))
	api.HandleH("POST /admin/llm-connections/{id}/test", admin(http.HandlerFunc(deps.AI.LLMConnection.TestConnection)))
}
